### PR TITLE
hooks: disable hooks by default from v0.14

### PR DESCRIPTION
**Pull request by fmuyassarov**
We have deprecated hooks in v0.12.0 but kept it enabled by default. Starting from v0.13.x we are starting to disable it by default and plan to fully remove it in v0.14.x if there are no objections from the community.

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
https://github.com/kubernetes-sigs/node-feature-discovery/issues/856

File: docs/usage/customization-guide.md
Comment (mythi) [inline docs/usage/customization-guide.md:281]: no objections for this PR but I find it surprising this change is done in the middle of `v0.13` cycle

Comment (ArangoGutierrez) [reply docs/usage/customization-guide.md:281]: Agree, please move to v0.14, the final deprecation will happen N+2 (v0.16)

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]
Could you add logs here 

https://github.com/kubernetes-sigs/node-feature-discovery/blob/54bd4c5d7437669511cf3639f0e325ca30b9eaa0/source/local/local.go#L115 
 similar to 

https://github.com/kubernetes-sigs/node-feature-discovery/blob/54bd4c5d7437669511cf3639f0e325ca30b9eaa0/pkg/nfd-worker/nfd-worker.go#L510 

Something along these lines (Feel free to enhance) :
```go
    klog.Warningf("Hooks are being DEPRECATED and will be removed in a future release.")
```

File: docs/usage/customization-guide.md
Comment (marquiz) [reply docs/usage/customization-guide.md:281]: Yeah, no earlier than v0.14. We definitely don't want to break existing users with a patch release

Comment (fmuyassarov) [reply docs/usage/customization-guide.md:281]: agreed, will move to v0.14

**Comment by fmuyassarov** [comment]
/hold 
until we reach v0.14 release planing.

**Comment by fmuyassarov** [comment]
/retitle hooks: disable hooks by default from v0.14

**Comment by ArangoGutierrez** [comment]
> fmuyassarov

@fmuyassarov rebase is needed

**Comment by fmuyassarov** [comment]
> > fmuyassarov
> 
> @fmuyassarov rebase is needed

Done

File: docs/reference/worker-configuration-reference.md
Comment (marquiz) [inline docs/reference/worker-configuration-reference.md:355]: We could document the default like the other options, i.e. add a line  `Default: false` above

File: source/local/local.go
Comment (marquiz) [inline source/local/local.go:116]: This message looks a bit strange, user might not be interested what's the default, but might be interested that they're deprecated

```suggestion
		klog.InfoS("NOTE: hooks are deprecated and will be completely removed in a future release.")
```

**Review by marquiz — COMMENTED** [review]
Should we get this merged? @ArangoGutierrez @fmuyassarov?

**Review by marquiz — COMMENTED** [review]
Should we get this merged? @ArangoGutierrez @fmuyassarov?

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm
unhold at will @marquiz

**Comment by marquiz** [comment]
Before merging, please @fmuyassarov see the unresolved comments

**Comment by fmuyassarov** [comment]
@marquiz Rebase is done now. PTAL, thanks.

**Review by marquiz — APPROVED** [review]
Thanks @fmuyassarov! One step closer to getting rid of hooks 😇

**Review by marquiz — APPROVED** [review]
Thanks @fmuyassarov! One step closer to getting rid of hooks 😇

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

**Pull request merged**
